dlt/common/runtime/typing.py

```diff
@@ -0,0 +1,18 @@
+from typing import Any, Callable, Dict, List, Literal, Optional, Sequence, Set, Type, TypedDict, NewType, Union, get_args
+
+
+TExecInfoNames = Literal["kubernetes", "docker", "codespaces", "github_actions", "airflow", "notebook", "colab","aws_lambda","gcp_cloud_function"]
+
+class TVersion(TypedDict):
+    """TypeDict representing a library version"""
+    name: str
+    version: str
+
+class TExecutionContext(TypedDict):
```

Comment:
Looking good! for open telemetry we can collect way more information that is not anonymous. but in other PR